docs/pipelines.md

```diff
@@ -362,6 +362,7 @@ There are a lot of scenarios where `WhenExpressions` can be really useful. Some
 - Checking if a git file has changed in the previous commits
 - Checking if an image exists in the registry
 - Checking if the name of a CI job matches
+- Checking if an optional Workspace has been provided
```

Comment:
I think we should add more detail that the workspace variable is available, in addition to Parameters and Results, in this part of the docs:
https://github.com/tektoncd/pipeline/blob/0b85cea1ff071f4769771a65b97d2684f73f2014/docs/pipelines.md#L324-L331

And include a snippet and reference the end-to-end example here:
https://github.com/tektoncd/pipeline/blob/0b85cea1ff071f4769771a65b97d2684f73f2014/docs/pipelines.md#L333-L355

Reply:
Nice thank you, I've updated the doc per your suggestions.